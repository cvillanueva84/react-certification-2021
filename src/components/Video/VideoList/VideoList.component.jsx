import React from 'react';
import { videoData } from '../../../youtube-videos-mock';
import { Container } from './VideoList.styled';
import VideoItem from '../VideoItem/VideoItem.component';

export default function VideoList() {
  const { items } = videoData;

  const avatar = items[0].snippet.thumbnails.default.url;

  return (
    <Container>
      {items.slice(1).map((item, id) => {
        const {
          snippet: {
            thumbnails: {
              medium: { url },
            },
          },
        } = item;

        const {
          snippet: { title },
        } = item;

        const {
          snippet: { description },
        } = item;

        return (
          <VideoItem
            key={id}
            url={url}
            title={title}
            description={description}
            avatar={avatar}
          />
        );
      })}
    </Container>
  );
}
