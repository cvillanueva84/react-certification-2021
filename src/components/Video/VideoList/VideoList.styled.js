import styled from 'styled-components';

export const Container = styled.section`
  margin-top: 100px;
  position: relative;
  z-index: 0;
  width: 800px;

  @media screen and (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 1.5rem;
  }

  @media screen and (min-width: 992px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;
