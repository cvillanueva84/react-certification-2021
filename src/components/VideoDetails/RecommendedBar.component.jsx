import React from 'react';
import { Link } from 'react-router-dom';
import { StyledRecommendedBar } from '../../styled/VideoDetails.styled';

function RecommnendedBar({ recommended }) {
  console.log(recommended);
  return (
    <StyledRecommendedBar>
      {recommended.map((item) => {
        const { snippet, id } = item;
        const { thumbnails, title } = snippet;
        const { videoId } = id;
        return (
          <Link to={`/${videoId}`}>
            <div className="recommended-box">
              <img alt="youtube video thumbnail" src={thumbnails.medium.url} />
              <p>{title}</p>
            </div>
          </Link>
        );
      })}
    </StyledRecommendedBar>
  );
}

export default RecommnendedBar;
