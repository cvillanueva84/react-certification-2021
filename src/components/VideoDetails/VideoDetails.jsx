import React from 'react';
import {
  Grid,
  GridItem,
  List,
  ListItem,
  ListItemAvatar,
  ListItemBody,
  Typography,
} from './VideoDetails.styled';

export const VideoDetails = ({ video, videoList = [], selectVideo }) => {
  const { id, snippet } = video;
  return (
    <Grid container>
      <GridItem xs={12} md={8}>
        <iframe
          src={`https://www.youtube.com/embed/${id.videoId}`}
          frameBorder="0"
          title="video"
          width="100%"
          height="600px"
        />
        <Typography as="h2">{snippet.title}</Typography>
        <p>{snippet.description}</p>
      </GridItem>
      <GridItem xs={12} md={4}>
        <List>
          {videoList.map((item) => (
            <ListItem
              key={item.id.videoId}
              type="button"
              onClick={() => selectVideo(item)}
            >
              <ListItemAvatar
                src={item.snippet.thumbnails.medium.url}
                alt={item.snippet.title}
              />
              <ListItemBody>
                <h5>{item.snippet.title}</h5>
                <p>{item.snippet.channelTitle}</p>
              </ListItemBody>
            </ListItem>
          ))}
        </List>
      </GridItem>
    </Grid>
  );
};
