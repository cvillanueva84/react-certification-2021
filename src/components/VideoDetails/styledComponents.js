import styled from 'styled-components';

export const TitleSpan = styled.span`
  font-size: 3rem;
  font-weight: bold;
`;
TitleSpan.displayName = 'TitleSpan';

export const DescriptionSpan = styled.span`
  font-size: 1.6rem;
`;
DescriptionSpan.displayName = 'DescriptionSpan';