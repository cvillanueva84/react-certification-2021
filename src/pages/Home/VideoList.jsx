import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 