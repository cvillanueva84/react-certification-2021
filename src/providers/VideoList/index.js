export { default, useAuth } from './Search.provider';
