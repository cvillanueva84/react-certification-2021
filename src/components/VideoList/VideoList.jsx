import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
