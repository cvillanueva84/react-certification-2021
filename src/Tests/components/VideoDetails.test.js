import React from 'react';
import { mount } from 'enzyme/build';
import VideoDetails from '../../components/VideoDetails/VideoDetails.component';
import { MemoryRouter, Route } from 'react-router';
import { videoList as videos } from '../mockData/videosList'
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';

describe('Tests of the VideoDetials component', () => {
    const wrapper = mount( 
        <VideoListContext.Provider
            value={{videos}}
        >
            <MemoryRouter
                initialEntries={['/video/Po3VwR_NNGk']}
                >
                <Route path="/video/:id" >
                    <VideoDetails />
                </Route> 
            </MemoryRouter>
        </VideoListContext.Provider>)

       test('should display the component correctly', () => {
         expect(wrapper).toMatchSnapshot()  
       })
        

})
