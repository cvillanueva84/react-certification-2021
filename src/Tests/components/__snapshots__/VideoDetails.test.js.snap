// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the VideoDetials component should display the component correctly 1`] = `
<div
  className="video-details-container"
>
  <div
    className="video-details-selected-video"
  >
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      src="https://www.youtube.com/embed/Po3VwR_NNGk"
      title="test"
    />
  </div>
  <div
    className="video-details-similar-videos"
  >
    <div
      className="recommended-video__container"
    >
      <a
        href="/video/UCPGzT4wecuWM0BH9mPiulXg"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="recommended-video__card"
        >
          <div
            className="recommended-video__card-img"
          >
            <img
              alt="Wizeline"
              src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="recommended-video__card-text"
          >
            <span
              className="sc-dlnjwi jkBdIj"
            >
              Wizeline
            </span>
          </div>
        </div>
      </a>
      <a
        href="/video/nmXMgqjQzls"
        onClick={[Function]}
        style={
          Object {
            "color": "#000",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="recommended-video__card"
        >
          <div
            className="recommended-video__card-img"
          >
            <img
              alt="Video Tour | Welcome to Wizeline Guadalajara"
              src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="recommended-video__card-text"
          >
            <span
              className="sc-dlnjwi jkBdIj"
            >
              Video Tour | Welcome to Wizeline Guadalajara
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
